Cabal/tests/UnitTests/Distribution/Simple/Utils.hs

```diff
@@ -84,6 +85,10 @@ rawSystemStdInOutTextDecodingTest ghcPath
     Left err | isDoesNotExistError err -> Exception.throwIO err -- no ghc!
              | otherwise               -> return ()
 
+dropExeExtensionTest :: Assertion
+dropExeExtensionTest =
+  assertBool "dropExeExtension didn't drop exeExtension!" $
+    dropExeExtension ("foo" ++ exeExtension) == "foo"
```

Comment:
you don't use `<.>` here?

Reply:
I suppose I could use `<.>` if you prefer? Since it swallows the second dot anyways.

Reply:
You **should**. You are adding extension to a `FilePath`, even it's a `String` (there's a dream that we could move to opaque path type)

Reply:
Haha, sure :)

Reply:
TL;DR


IMHO you should use `Distribution.Simple.BuildPaths.exeExtension Distribution.Simple.buildPlatform`

and avoid CPP